core/platform/CMakeLists.txt

```diff
@@ -14,18 +14,31 @@ set(LF_PLATFORM_FILES
     lf_rp2040_support.c
 )
 
-if(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
```

Comment:
A long long time ago we talked about moving as much of this conditional compilation stuff as possible from CMake to preprocessor. But recently it seems like we have changed our minds and decided that CMake is powerful enough for us to use it for everything, in which case I'm happy having this code here.

Reply:
You are right. I could add both files to the build and use macros. Or do `#include "lf_atomic32.c` (or 64) from the different platform implementations.

Personally I would prefer to just stick to CMake. But I understand that there might be use-cases down the line where CMake is not an option... so yeah.